prober/utils_test.go

```diff
@@ -162,14 +163,25 @@ func TestChooseProtocol(t *testing.T) {
 	registry = prometheus.NewPedanticRegistry()
 
 	ip, _, err = chooseProtocol(ctx, "ip4", false, "ipv6.google.com", registry, logger)
-	if err != nil && err.Error() != "unable to find ip; no fallback" {
+	if err != nil && !strings.HasPrefix(err.Error(), "unable to find ip; no fallback") {
 		t.Error(err)
 	} else if err == nil {
 		t.Error("should set error")
 	}
 	if ip != nil {
 		t.Error("without fallback it should not answer")
 	}
+
+	registry = prometheus.NewPedanticRegistry()
+	ip, _, err = chooseProtocol(ctx, "ip4", true, "does-not-exist.google.com", registry, logger)
```

Comment:
It's probably better to work from example.com, less chance of such a subdomain popping up.